WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -20,49 +23,51 @@
 
 import javax.inject.Inject;
 
-public class PostsListActivity extends AppCompatActivity {
+public class PostsListActivity extends AppCompatActivity implements SearchView.OnQueryTextListener {
+    public static final String EXTRA_SEARCH_TERM = "searchTerm";
     public static final String EXTRA_VIEW_PAGES = "viewPages";
     public static final String EXTRA_ERROR_MSG = "errorMessage";
 
     private boolean mIsPage = false;
     private PostsListFragment mPostList;
     private SiteModel mSite;
 
+    private MenuItem mSearchAction;
+    private String mCurrentSearch;
+
     @Inject SiteStore mSiteStore;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         ((WordPress) getApplication()).component().inject(this);
 
-        setContentView(R.layout.post_list_activity);
+        // init
+        unpackIntent(getIntent());
+        if (savedInstanceState != null) {
+            mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
+        }
 
-        mIsPage = getIntent().getBooleanExtra(EXTRA_VIEW_PAGES, false);
+        // need a Site to continue
+        if (mSite == null) {
+            ToastUtils.showToast(this, R.string.blog_not_found, ToastUtils.Duration.SHORT);
+            finish();
+            return;
+        }
 
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
+        // set layout and init toolbar
+        setContentView(R.layout.post_list_activity);
 
+        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
         ActionBar actionBar = getSupportActionBar();
         if (actionBar != null) {
-            actionBar.setTitle(getString(mIsPage ? R.string.pages : R.string.posts));
+            setTitle(getString(mIsPage ? R.string.pages : R.string.posts));
```

Comment:
Another nitpick: It looks like we are calling this exact same `setTitle` twice. Do you think it'd be better to remove the `public void setTitle(CharSequence title) {` override and add a helper method for setting the title?